lua/cfc_random_spawn/module/sv_player_spawning.lua

```diff
@@ -325,18 +389,37 @@ function CFCRandomSpawn.getOptimalSpawnPos()
     return randomFreeSpawn.spawnPos, randomFreeSpawn.spawnAngle
 end
 
+function CFCRandomSpawn.getMostEnclosedSpawnPos()
+    if not CFCRandomSpawn.spawnsUnderARoof then
+        CFCRandomSpawn.spawnsUnderARoof = getEnclosedSpawns( customSpawnsForMap )
+    end
+
+    local mostEnclosedSpawn = getMostEnclosedSpawn( CFCRandomSpawn.spawnsUnderARoof )
+    return mostEnclosedSpawn.spawnPos, mostEnclosedSpawn.spawnAngle
+end
+
 function CFCRandomSpawn.handlePlayerSpawn( ply )
     if not mapHasCustomSpawns then return end
     if not ( ply and IsValid( ply ) ) then return end
     if IsValid( ply.LinkedSpawnPoint ) then return end
 
-    local optimalSpawnPosition, optimalSpawnAngles = CFCRandomSpawn.getOptimalSpawnPos()
+    timer.Simple( 0, function()
```

Comment:
this might break the spawnpoint entities, make sure that it doesnt

Reply:
It has the if IsValid( ply.LinkedSpawnPoint ) then return end but ill double check

Reply:
Oh i missed that, yeah should be good